packages/lib/src/analytics/types.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * `ActionTypePrefixes` is a union type of all the prefixes that can be used to create an `ActionType`.
+ */
+type ActionTypePrefixes =
+  | "Changed"
+  | "Clicked"
+  | "Created"
+  | "Deleted"
+  | "Redirected"
+  | "Filtered"
+  | "Saved"
+  | "Sorted"
+  | "Toggled"
+  | "Viewed"
+  | "Used"
+  | "System Event";
+
+/**
+ * `ActionType` is a type that represents an action that can be sent to our analytics provider.
+ * @param name - The name of the action to send to our analytics provider
+ * @example - { name: "Clicked Button" }
+ * @example - { name: "Changed Input" }
+ */
+export interface ActionType {
+  name: `${ActionTypePrefixes}${string}`;
+}
+
+/**
+ * `sendEvent` is the function call to send an analytics event. It requires an Action
+ */
+type sendEvent<Action extends ActionType> = (action: Action) => void;
+
+/**
+ * `AnalyticsProperties` is an object that represents the properties and additional metadata to send with an event to our analytics provider.
+ *
+ */
+export interface AnalyticsProperties {
+  [key: string]: string | number | boolean;
+}
+
+/**
+ * DO NOT USE THIS IN YOUR CODE
+ *
+ * This is simply a clever named type to help with the typing of the `useAnalyticsRoot` hook.
+ * `AnalyticsIdentifier` is a string that represents the object to send with an event to our analytics provider.
+ * @example - "Page Name"
+ * @example - "Component Name"
+ */
+export type AnalyticsIdentifier = string;
```

Comment:
Should we maybe include this type in the hooks file with no export, or just type the `useAnalyticsRoot` hook with `Identifier extends string`? Would be nice to prevent it from being used in code instead of relying on a comment to explain.

Reply:
Great catch!